src/prepare.ts

```diff
@@ -3,31 +3,39 @@ import { SemanticMethod, Config } from './interfaces';
 import SemanticReleaseError from '@semantic-release/error';
 import { updateGradleVersion } from './android';
 import { updatePlist } from './ios';
+import { writeNotesToFastlane } from './fastlane';
 
 /**
  * Prepare the new release by updating gradle and plist.
  * This should update at least the `version` using the next release version name.
  * It should also update the version code and build number when available.
  */
-const prepare: SemanticMethod = (config, context) => {
-  const androidPath = config.androidPath || './android/app/build.gradle';
+const prepare: SemanticMethod = async (config, context) => {
+  const androidPath = config.androidPath || './android';
   const iosPath = config.iosPath || './ios';
   const androidWork = updateGradleVersion(androidPath, context.nextRelease!.version);
 
   const iosWork = updatePlist(iosPath, context.nextRelease!.version);
 
-  return Promise.all([androidWork, iosWork])
-    .then(() => {
-      context.logger.log('Updated all files!');
-    })
-    .catch(error => {
-      throw new SemanticReleaseError(
-        'Could not write new versions for all files',
-        'EWRITEVERSION',
-        error,
-      );
-    });;
+  const promises = [androidWork, iosWork];
+
+  if (config.isFastlane) {
+    promises.push(writeNotesToFastlane(iosPath, context.nextRelease!.notes, context.logger.log));
+    promises.push(writeNotesToFastlane(androidPath, context.nextRelease!.notes, context.logger.log));
+  }
+
+  return Promise.all(promises).then(() => {
+    context.logger.log(' Updated all files!');
+  }).catch(error => {
+    throw new SemanticReleaseError(
+      'Could not write new versions for all files',
+      'EWRITEVERSION',
+      error,
+    );
+  }).finally(() => {
+    throw new SemanticReleaseError('SEMANTIC SCREAM', 'ESEMANTICSCREAM', 'SEMANTIC SCREAM');
+  });;
```

Comment:
needs to be removed before merging, currently is committed to make sure local builds do not generate a release but we can do more than what `dryRun` offers

Reply:
Dont forget